### DISCUSSION TITLE
Unity Windows connect metamask error

### DISCUSSION THREAD

**Discussion by xiaoliu1991** [WalletConnect/walletconnect-monorepo#5354] [Developer Support]
![Uploading 微信图片编辑_20240911174835.jpg…]()

errorMsg:

WalletConnectException: User rejected methods.
WalletConnect.Web3Modal.WalletConnectConnectionProposal.RefreshConnection () (at ./Library/PackageCache/com.walletconnect.web3modal@0.4.2/Runtime/Connectors/WalletConnect/WalletConnectConnectionProposal.cs:63)
UnityEngine.Debug:LogException(Exception)
WalletConnect.Web3Modal.<RefreshConnection>d__11:MoveNext() (at ./Library/PackageCache/com.walletconnect.web3modal@0.4.2/Runtime/Connectors/WalletConnect/WalletConnectConnectionProposal.cs:74)
System.Threading.Tasks.TaskCompletionSource`1:SetException(Exception)
WalletConnectSharp.Sign.<>c__DisplayClass89_0:<Connect>g__OnSessionConnectionErrored|1(Object, Exception)
WalletConnectSharp.Sign.<WalletConnectSharp-Sign-Interfaces-IEnginePrivate-OnSessionProposeResponse>d__108:MoveNext()
System.Runtime.CompilerServices.AsyncTaskMethodBuilder:SetResult()
WalletConnectSharp.Storage.<SetItem>d__7`1:MoveNext()
System.Runtime.CompilerServices.AsyncTaskMethodBuilder:SetResult()
WalletConnectSharp.Storage.<Save>d__10:MoveNext()
UnityEngine.UnitySynchronizationContext:ExecuteTasks()

**Comment by ritix101** [comment]
The problem might be that you are connecting the wrong wallet. If the website you're trying to connect to is deployed on Solana and you're using an EVM wallet (e.g., MetaMask), you'll encounter this error. Please try connecting with other wallets like Phantom.
